index.js

```diff
@@ -1,58 +1,64 @@
+    this.renderBookList();
+  }
 
-// Function to remove a book from the collection
-function removeBook(index) {
-  books = books.filter((books, i) => i.toString() !== index);
-  localStorage.setItem('books', JSON.stringify(books));
-}
+  handleAddBook(event) {
+    event.preventDefault();
+    const title = this.titleInput.value;
+    const author = this.authorInput.value;
+    this.addBook(title, author);
+    this.renderBookList();
+    this.resetForm();
+  }
+
+  addBook(title, author) {
+    this.books.push({ title, author });
+    this.updateLocalStorage();
+  }
+
+  removeBook(index) {
+    this.books = this.books.filter((book, i) => i !== index);
+    this.updateLocalStorage();
+  }
+
+  renderBookList() {
+    this.bookList.innerHTML = '';
+    this.books.forEach((book, index) => {
+      const li = document.createElement('p');
+      li.innerHTML = `${book.title} by ${book.author} <br> `;
+      this.bookList.appendChild(li);
+      li.dataset.index = index;
 
-// Function to render the book list
-function renderBookList() {
-  bookList.innerHTML = '';
-  books.forEach((book, index) => {
-    const li = document.createElement('p');
-    li.innerHTML = `${book.title} by ${book.author} <br> `;
-    bookList.appendChild(li);
-    li.dataset.index = index;
-    const p = document.createElement('hr');
-
-    // add a remove button for each book
-    const removeButton = document.createElement('button');
-    removeButton.innerHTML = 'Remove';
-    li.appendChild(removeButton);
-    removeButton.addEventListener('click', (event) => {
-      const { index } = event.target.parentNode.dataset;
-      removeBook(index);
-      renderBookList();
+      const hr = document.createElement('hr');
+      li.appendChild(hr);
+
+      const removeButton = document.createElement('button');
+      removeButton.innerHTML = 'Remove';
+      removeButton.addEventListener('click', () => {
+        const index = parseInt(li.dataset.index, 10);
+        this.removeBook(index);
+        this.renderBookList();
+      });
+      li.appendChild(removeButton);
     });
-    li.appendChild(p);
-  });
-}
+  }
 
-// Function to add a new book to the collection
-function addBook(title, author) {
-  books.push({ title, author });
-  localStorage.setItem('books', JSON.stringify(books));
-}
+  resetForm() {
+    this.addBookForm.reset();
+  }
 
-// Event listener for the add book form submission
-addBookForm.addEventListener('submit', (event) => {
-  event.preventDefault();
-  const title = addBookForm.title.value;
-  const author = addBookForm.author.value;
-  addBook(title, author);
-  renderBookList();
-  addBookForm.reset();
-});
-
-// Render the book list on page load
-renderBookList();
-
-document.querySelector('form').addEventListener('submit', () => {
-  this.reset();
-});
\ No newline at end of file
+  updateLocalStorage() {
+    localStorage.setItem('books', JSON.stringify(this.books));
+  }
```

Comment:
According to the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m2_plain_js_classes_v1_1.md), it is important that we maintain all of the original functionality of the application (i.e. no bugs were introduced while refactoring). 👀

Kindly ensure that data is preserved in the browser's memory by using [localStorage](https://github.com/microverseinc/curriculum-html-css/blob/main/local_storage.md). 🤗👍